scripts/irods/test/test_iput_options.py

```diff
@@ -195,6 +197,33 @@ def test_iput_symlink_to_self_on_commandline_4016(self):
             self.user0.assert_icommand_fail( cmd, 'STDOUT_SINGLELINE', symtoself );
             self.user0.assert_icommand_fail( cmd, 'STDOUT_SINGLELINE', goodfile );
 
+    def test_iput_recursive_bulk_upload__4657(self):
+        bulk_str = 'Bulk upload'
+        source_path = tempfile.mkdtemp()
+        single_file = tempfile.mkstemp(dir=source_path)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [local variable 'single_file' is assigned to but never used](https://app.codacy.com/manual/trel/irods/pullRequest?prid=4893066)